lexer/impl/lexer.cpp

```diff
@@ -1,6 +1,312 @@
+                    break;
+
+                default:
+
+                    if (std::isdigit(c))
+                    { // if number constant
+                        index--;
+                        std::string num = readWhileNum(index, word);
+                        TokenType type = TokenType::IntegerLiteral;
+                        if (index < word.length() && word[index] == '.')
+                        { // if real
+                            int oldIndex = index;
+                            index++;
+                            std::string floatPart = readWhileNum(index, word);
+                            if (floatPart.length() == 0)
+                            { //eg ..
+                                index = oldIndex;
+                                result.push_back({type, lineNum, num});
+                                break;
+                            }
+                            num.append(".");
+                            num.append(floatPart);
+                            type = TokenType::RealLiteral;
+                        }
+                        result.push_back({type, lineNum, num});
+                    }
+                    else if (std::isalpha(c))
+                    { // if var name
+                        std::string name = toStr(c);
+                        while (index < word.length() && (std::isalpha(word[index]) || std::isdigit(word[index]) || word[index] == '_'))
+                        {
+                            name.push_back(word[index++]);
+                        }
+                        result.push_back({TokenType::Identifier, lineNum, name});
+                    }
+                    else
+                    { // non-valid construction
+                        // std::cout << "INVALID CHAR" << c << "\n";
+                        result.push_back({TokenType::InvalidToken, lineNum, toStr(c)});
+                    }
+
+                    break;
+                }
+            }
+        }
+        return 0;
+    }
+
+    /**
+     * Processing a line of riddle code. It updates the 'result' vector of tokens
+     * 
+     * @param line, string with line
+    */
+    void processLine(std::string line)
+    {
+        lineNum++;
+        std::istringstream iss(line);
+        std::string word;
+        while (std::getline(iss, word, ' '))
+        {
+            if (processWord(word) == -1)
+            { //if get a single-line comment
+                return;
+            }
+        }
+        result.push_back({TokenType::NewLine, lineNum, "\n"});
+    }
+    // ==================================PUBLIC================================
+    /**
+ * Gets the trie from the lexer's namespace. 
+ * Method should be used for testing and debugging only
+ * 
+ * @return trie, used during parsing (debugging purposes)
+*/
+    common::Trie<TokenType> getTrie()
+    {
+        initTrie();
+        return trie;
+    }
```

Comment:
So, yeah, this is also 'rudimentary' since this function is as same as `lexer::initTrie`